scipy/stats/_stats_py.py

```diff
@@ -9440,3 +9442,88 @@ def rankdata(a, method='average', *, axis=None, nan_policy='propagate'):
         result[nan_indexes] = np.nan
 
     return result
+
+
+# TODO: axis parameter
+# TODO: add _axis_nan_policy_factory decorator
```

Comment:
I'd recommend against including `axis` and `nan_policy` in the initial scope. Let's make this PR as simple as possible to sped up review, and these other options can be added later. When you do look into that, please use `ttest_1samp` as an example. I think we can treat `alpha` as we do `popmean`.